File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.js
var maxScore = function(C, K) {
    let total = 0
    for (let i = 0; i < K; i++) total += C[i]
    let best = total
    for (let i = K - 1, j = C.length - 1; ~i; i--, j--)
        total += C[j] - C[i], best = Math.max(best, total)
    return best
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int total = 0;
        for(int i=0;i<k;i++)
            total += cardPoints[i];
        int best = total;
        for(int i=k-1,j=cardPoints.length-1;i>=0;i--,j--)
        {
            total += cardPoints[j]-cardPoints[i];
            best = Math.max(best,total);
        }
        return best;
    }
}